bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/discovery/HomeConnectDiscoveryService.java

```diff
@@ -0,0 +1,170 @@
+
+    private static final int SEARCH_TIME = 20;
+
+    private final HomeConnectBridgeHandler bridgeHandler;
+    private final Logger logger;
+
+    /**
+     * Construct an {@link HomeConnectDiscoveryService} with the given
+     * {@link org.openhab.core.thing.binding.BridgeHandler}.
+     *
+     * @param bridgeHandler bridge handler
+     */
+    public HomeConnectDiscoveryService(HomeConnectBridgeHandler bridgeHandler) {
+        super(DISCOVERABLE_DEVICE_THING_TYPES_UIDS, SEARCH_TIME, true);
+        this.bridgeHandler = bridgeHandler;
+        logger = LoggerFactory.getLogger(HomeConnectDiscoveryService.class);
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Starting device scan.");
+
+        HomeConnectApiClient apiClient = bridgeHandler.getApiClient();
+
+        try {
+            List<HomeAppliance> appliances = apiClient.getHomeAppliances();
+            logger.debug("Scan found {} devices.", appliances.size());
+
+            // add found devices
+            for (HomeAppliance appliance : appliances) {
+                @Nullable
+                ThingTypeUID thingTypeUID = getThingTypeUID(appliance);
+
+                if (alreadyExists(appliance.getHaId())) {
+                    logger.debug("Device {} ({}) already added as thing.", appliance.getHaId(),
+                            appliance.getType().toUpperCase());
+                } else if (thingTypeUID != null && DISCOVERABLE_DEVICE_THING_TYPES_UIDS.contains(thingTypeUID)) {
+                    logger.info("Found {} ({}).", appliance.getHaId(), appliance.getType().toUpperCase());
+                    bridgeHandler.getThing().getThings().forEach(thing -> thing.getProperties().get(HA_ID));
+
+                    Map<String, Object> properties = new HashMap<>();
+                    properties.put(HA_ID, appliance.getHaId());
+                    String name = appliance.getBrand() + " " + appliance.getName() + " (" + appliance.getHaId() + ")";
+
+                    DiscoveryResult discoveryResult = DiscoveryResultBuilder
+                            .create(new ThingUID(BINDING_ID, appliance.getType(),
+                                    bridgeHandler.getThing().getUID().getId(), appliance.getHaId()))
+                            .withThingType(thingTypeUID).withProperties(properties)
+                            .withBridge(bridgeHandler.getThing().getUID()).withLabel(name).build();
+                    thingDiscovered(discoveryResult);
+                } else {
+                    logger.debug("Ignoring unsupported device {} of type {}.", appliance.getHaId(),
+                            appliance.getType());
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Exception during scan.", e);
+        }
+        logger.debug("Finished device scan.");
+    }
+
+    @Override
+    public void deactivate() {
+        super.deactivate();
+        removeOlderResults(new Date().getTime());
```

Comment:
`java.time` API would be better, not that it would make a lot of difference here.

Reply:
Changed it to `System.currentTimeMillis()`